packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/EditTab.tsx

```diff
@@ -101,55 +98,80 @@ export const EditTab: React.FC = () => {
 
     const stepTitle = data.steps.find(step => step.id === isEditStep.id)?.title || "";
 
-    const handleStepMove = (source: any, step: FbFormStep): void => {
-        const { pos, formStep } = source;
-
-        if (pos) {
-            if (pos.index === null) {
+    const handleStepMove = (item: DragObjectWithFieldInfo, formStep: FbFormStep) => {
+        if (item.pos) {
+            if (item.pos.index === null) {
                 return;
             }
         }
 
-        moveStep({
-            step,
-            formStep
-        });
+        const moveStepParams: MoveStepParams = {
+            target: {
+                containerId: item.container?.id || "",
+                position: item.pos
+            },
+            destination: {
+                containerId: formStep.id
+            }
+        };
+
+        moveStep(moveStepParams);
+
+        return undefined;
     };
 
     // This function will render drop zones on the top of the step,
     // if steps are locatted above "source" ("source" step is the step that we move).
-    const renderTopDropZone = (sourceStepId: string | undefined, targetStepId: string) => {
-        if (!sourceStepId) {
+    const renderTopDropZone = (item: IsVisibleCallableParams, targetStepId: string) => {
```

Comment:
Let's use `useCallback` here if possible.

Reply:
@adrians5j

Also I have noticed that if we add a `useCallback` or `useMemo` methods from `useFormStep` stops working properly.
For instance moveRow does not work properly, it would only move one field and delete another field instead of moving an entire row into another step. 
And it is only one of many bugs that occurs after using `useCallback`.
 And I just want to know whether it is okay that we would not use `useCallback` and `useMemo`?